25.0.0.7-beta staging (#4419)
OpenLiberty/blogs: posts/2025-07-01-25.0.0.7-beta.adoc

---
layout: post

> title: "Microprofile 7.1 and updates to its component specifications in 25.0.0.7-beta"

Reviewer: ```suggestion
title: "MicroProfile 7.1 and updates to its component specifications in 25.0.0.7-beta"
```

# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/navaneethsnair1
author_github: https://github.com/navaneethsnair1

> seo-title: Microprofile 7.1 and updates to its component specifications in 25.0.0.7-beta - OpenLiberty.io

Reviewer: ```suggestion
seo-title: MicroProfile 7.1 and updates to its component specifications in 25.0.0.7-beta - OpenLiberty.io
```

Reviewer 2: Thank you @Emily-Jiang

> seo-description: This beta release introduces support for Microprofile 7.1, which updates two component specifications- Microprofile Telemetry and Microprofile OpenAPI.

Reviewer: ```suggestion
seo-description: This beta release introduces support for MicroProfile 7.1, which updates two component specifications- MicroProfile Telemetry and MicroProfile OpenAPI.
```

> blog_description: This beta release introduces support for Microprofile 7.1, which updates two component specifications- Microprofile Telemetry and Microprofile OpenAPI.

Reviewer: ```suggestion
blog_description: This beta release introduces support for MicroProfile 7.1, which updates two component specifications- MicroProfile Telemetry and MicroProfile OpenAPI.
```

open-graph-image: https://openliberty.io/img/twitter_card.jpg
open-graph-image-alt: Open Liberty Logo
---

> = Microprofile 7.1 and updates to its component specifications in 25.0.0.7-beta

Reviewer: ```suggestion
= MicroProfile 7.1 and updates to its component specifications in 25.0.0.7-beta
```

Navaneeth S Nair <https://github.com/navaneethsnair1>
:imagesdir: /
:url-prefix:
:url-about: /
//Blank line here is necessary before starting the body of the post.

> This beta release introduces support for Microprofile 7.1, which updates two component specifications- Microprofile Telemetry and Microprofile OpenAPI.

Reviewer: ```suggestion
This beta release introduces support for MicroProfile 7.1, which updates two component specifications: MicroProfile Telemetry and MicroProfile OpenAPI.
```

// // // // // // // //
// Change the RELEASE_SUMMARY to an introductory paragraph. This sentence is really
// important because it is supposed to grab the readers attention.  Make sure to keep the blank lines
//
// Throughout the doc, replace 25.0.0.7-beta with the version number of Open Liberty, eg: 22.0.0.2-beta
// // // // // // // //

The link:{url-about}[Open Liberty] 25.0.0.7-beta includes the following beta features (along with link:{url-prefix}/docs/latest/reference/feature/feature-overview.html[all GA features]):

* <<microprofile, MicroProfile 7.1>>
* <<telemetry, MicroProfile Telemetry 2.1>>
* <<openapi, MicroProfile OpenAPI 4.1>>

See also link:{url-prefix}/blog/?search=beta&key=tag[previous Open Liberty beta blog posts].    
    
// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/31807
// Contact/Reviewer: Emily-Jiang
// // // // // // // // 
[#microprofile]
== MicroProfile 7.1

> MicroProfile 7.1 (`MP 7.1`) is a minor release. It has the following MicroProfile component specifications and works with Jakarta EE 10 Core Profile:

Reviewer 3: Instead of `It` say `MicroProfile 7.1`

Reviewer 2: @is273 wont it be repetitive to frame it 'MicroProfile 7.1 has the following MicroProfile component specifications and works with Jakarta EE 10 Core Profile:'

Reviewer 2: @is273 Can I frame it :
MicroProfile 7.1 has the following component specifications and works with Jakarta EE 10 Core Profile.

Still it sounds odd for me with the previous sentence

Reviewer 3: @navaneethsnair1 It may sound repetitive but we refrain from using "it" to refer to something.

Reviewer: I made the below suggestions and hopefully addressed both of your concerns

> MicroProfile 7.1 (`MP 7.1`) is a minor release. It has the following MicroProfile component specifications and works with Jakarta EE 10 Core Profile:

Reviewer: ```suggestion
MicroProfile 7.1 (`MP 7.1`) is a minor release, which has the following MicroProfile component specifications and works with Jakarta EE 10 Core Profile:
```

 
- link:https://jakarta.ee/specifications/coreprofile/10/[Jakarta EE 10 Core Profile]
- link:https://github.com/eclipse/microprofile-config/releases/tag/3.1[MicroProfile Config 3.1]
- link:https://github.com/eclipse/microprofile-fault-tolerance/releases/tag/4.1[MicroProfile Fault Tolerance 4.1]
- link:https://github.com/eclipse/microprofile-health/releases/tag/4.0[MicroProfile Health 4.0]
- link:https://github.com/eclipse/microprofile-telemetry/releases/tag/2.1[MicroProfile Telemetry 2.1]
- link:https://github.com/eclipse/microprofile-open-api/releases/tag/4.1[MicroProfile OpenAPI 4.1]
- link:https://github.com/eclipse/microprofile-jwt-auth/releases/tag/2.1[MicroProfile JWT Authentication 2.1]
- link:https://github.com/eclipse/microprofile-rest-client/releases/tag/4.0[MicroProfile Rest Client 4.0]

> In this, MicroProfile Telemetry 2.1 and MicroProfile OpenAPI 4.1 have updates while the other three specifications remain unchanged.

Reviewer: ```suggestion
In this releaes, MicroProfile Telemetry 2.1 and MicroProfile OpenAPI 4.1 have updates while the other five specifications remain unchanged.
```

Reviewer 2: Thank you @Emily-Jiang

> To enable MicroProfile 7.1, add the feature to your `server.xml`: 

Reviewer: ```suggestion
To enable MicroProfile 7.1, add the following feature to your `server.xml`: 
```

```xml
<featureManager>
    <feature>microProfile-7.1</feature>
</featureManager>
```

// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 

// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/31805
// Contact/Reviewer: yasmin-aumeeruddy
// // // // // // // // 
[#telemetry]
== MicroProfile Telemetry 2.1

> MicroProfile Telemetry 2.1(`mpTelemetry-2.1`) provides developers with updated Open Telemetry technology as the feature now uses OpenTelemetry-1.48.0, which was updated from 1.39.0. APIs are now stable.

Reviewer: ```suggestion
MicroProfile Telemetry 2.1(`mpTelemetry-2.1`) provides developers with updated Open Telemetry technology as the feature now uses OpenTelemetry v1.48.0, which was updated from v1.39.0. The previous exposed experimental APIs are now stable.
```

```xml
<featureManager>
   <feature>mpTelemetry-2.1</feature>
</featureManager>
```

If third-party API visibility was enabled for no other purpose than to manually instrument the application code to collect traces or to define custom metrics with versions of mpTelemetry before 2.1, it is recommended to remove them from the `server.xml`:

```xml
<webApplication id="app-name" location="app-name.war">       
   <classloader apiTypeVisibility="+third-party"/>
</webApplication>

> ```

Reviewer 4: ```suggestion
To use the OpenTelemetry API, it is no longer required to add the `apiTypeVisibility="+third-party"` attribute to the `classloader` element of the server.xml. This also means that it is now possible to deploy an application that uses the OpenTelemetry API by dropping it into the dropins directory.
```

For more information about MicroProfile Telemetry, see the following links:

- https://github.com/eclipse/microprofile-telemetry[MicroProfile Telemetry]
- https://github.com/open-telemetry/opentelemetry-specification[specification]
- https://openliberty.io/docs/latest/docs/latest/microprofile-telemetry.html[documentation]
    
    
// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 

// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/31811
// Contact/Reviewer: Azquelt
// // // // // // // // 
[#openapi]
== MicroProfile OpenAPI 4.1

OpenAPI is a standardized way of documenting REST APIs in a JSON or YAML format. MicroProfile OpenAPI helps to generate and serve OpenAPI documentation for REST applications that are built by using JAX-RS or Jakarta restfulWS. This can be useful for developers to test out the API during development, or for people who uses API in production.

> In this beta, the new MicroProfile OpenAPI 4.1 is introduced. To use it, first add the feature to your `server.xml`:

Reviewer 3: Instead of `it` say `MicroProfile OpenAPI 4.1`

Reviewer 5: I don't like the repetition suggested by @is273, how about this instead:

```suggestion
To use the new MicroProfile OpenAPI 4.1 feature, first add the feature to your `server.xml`:
```

Reviewer 2: What shall I finalize @is273 @Azquelt

```xml
<featureManager>
    <feature>mpOpenAPI-4.1</feature>
</featureManager>
```

> Then, deploy rest application and then see `/openapi` on liberty server to view the OpenAPI documentation, or go to `/openapi/ui` for a user interface that lets to browse through it. To enhance the documentation, use link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/[annotations] in code to add things like textual descriptions to each method. The annotations are available as a maven dependency:

Reviewer 3: Instead of 
"Then, deploy rest application and then see `/openapi` on liberty server to view the OpenAPI documentation, or go to `/openapi/ui` for a user interface that lets to browse through it." 
say 
"Deploy the rest application. Then see `/openapi` on the Liberty server to view the OpenAPI documentation, or go to `/openapi/ui` for a user interface to browse through the documentation."

Reviewer 5: If you're going to take out "Then" here, then I think you should also take out "First" in the previous paragraph.

Reviewer 5: I have to be honest, I don't understand these suggestions at all. This is a sequence of steps which follows on from the previous paragraph, why are we removing "then" from the start of one of the steps but leaving it on the other?

Reviewer 5: Ok, here's my suggestion:

"Next deploy any rest application. Then, in a web browser, go to `/openapi` on the Liberty server to view the YAML OpenAPI document, or go to `/openapi/ui` for a user interface to browse through the documentation."

- At this point in the instructions, we haven't talked about a rest application, so we can't say "the" rest application. One of the main points about `mpOpenAPI` is that it works with any rest application and we really want the user to try this out with their own application, but any will do for a demonstration.
- "then see `/openapi`" isn't very clear. We want the user to open a web browser and go to "localhost:9080/openapi", but we don't know if they're running their server locally.
- When reading this back, I didn't think it was particularly clear what the difference between the two endpoints was, so let's make it clearer that one is a yaml document, and the other is a UI.

Reviewer 2: How should I make the final edits @Azquelt @is273 ?

```xml
<dependency>
    <groupId>org.eclipse.microprofile.openapi</groupId>
    <artifactId>microprofile-openapi-api</artifactId>
    <version>4.1</version>
    <scope>provided</scope>
</dependency>
```

> MicroProfile OpenAPI 4.1 is a very small release, which adds:

Reviewer 3: Get rid of "very" and the comma

Reviewer 5: Should we not call it a very small release? It would be harder for it to be smaller, the only reason for this release is that we missed one property from the API in 4.0.

Reviewer 3: @Azquelt We try not to use words like "really" which seems like a filler word. To a user, what is the difference between a very small release and a small release.

Reviewer 5: Hmm, ok. That makes sense, this can be changed as you suggested.

My thinking was that users mostly want to know whether there's anything of interest to them, and in this case I expect this release to affect no-one at all, hence very small, it's basically a bug fix in the spec. But you're right, users are going to have to read for themselves what has changed, so saying small vs. very small doesn't accomplish much.

- The ability to set the link:https://spec.openapis.org/oas/v3.1.0.html#fixed-fields[`jsonSchemaDialect` field on the OpenAPI object] using the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/org.eclipse.microprofile.openapi/org/eclipse/microprofile/openapi/models/OpenAPI.html#setJsonSchemaDialect(java.lang.String)[model API]
- New methods `hasExtension` and `getExtension` on link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/org.eclipse.microprofile.openapi/org/eclipse/microprofile/openapi/models/Extensible.html[`Extensible`] in the model API

> More information about MicroProfile OpenAPI can be found in
> 
> - the liberty link:https://openliberty.io/guides/microprofile-openapi.html[guide]
> - the liberty link:https://openliberty.io/docs/latest/documentation-openapi.html[documentation]
> - the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/[javadoc]
> - the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/microprofile-openapi-spec-4.1.html[specification]

Reviewer 5: I think I should have put "Open Liberty" here instead of "liberty"
```suggestion
More information about MicroProfile OpenAPI can be found in

- the Open Liberty link:https://openliberty.io/guides/microprofile-openapi.html[guide]
- the Open Liberty link:https://openliberty.io/docs/latest/documentation-openapi.html[documentation]
- the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/[javadoc]
- the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/microprofile-openapi-spec-4.1.html[specification]
```
